src/app/api/users/users.service.ts

```diff
@@ -65,10 +66,25 @@ class UsersService extends BaseService {
    * @param keyword
    * @returns
    */
-  async getFilteredUsersStartingWith(keyword: string, limit?: number, nextToken?: string) {
+  async getFilteredUsersStartingWith(keyword: string, limit?: number, nextToken?: string, userType?: string) {
     const filterByKeyword = (users: FilterableUser[]) => filterUsersByKeyword(users, keyword)
-
     const { internalUsers, clients, companies } = await this.getGroupedUsers(limit || 500, nextToken)
+
```

Comment:
We are still hitting all 3 endpoints for IU, client and company, even if we only want IU for `FilterOptionsKeywords.TEAM`, and only client and company for `FilterOptionsKeywords.CLIENTS`